File: src/main/java/kr/co/lunatalk/domain/image/repository/ImageRepositoryCustom.java
package kr.co.lunatalk.domain.image.repository;

import kr.co.lunatalk.domain.image.domain.Image;
import kr.co.lunatalk.domain.image.domain.ImageType;

import java.util.List;

public interface ImageRepositoryCustom {
    List<Image> fetchProductImagesByProductId(Long productId);
    List<Image> fetchProductImagesByProductIds(List<Long> productIds);
	List<Image> findAllByReferenceIdAndImageType(Long referenceId, ImageType imageType);


}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/image/repository/ImageRepositoryCustom.kt
package kr.co.lunatalk.domain.image.repository

import kr.co.lunatalk.domain.image.domain.Image
import kr.co.lunatalk.domain.image.domain.ImageType

interface ImageRepositoryCustom {
    fun fetchProductImagesByProductId(productId: Long): List<Image>
    fun fetchProductImagesByProductIds(productIds: List<Long>): List<Image>
    fun findAllByReferenceIdAndImageType(referenceId: Long, imageType: ImageType): List<Image>
}
